### PR TITLE
Workflow: Update Python and Actions + Black formatting Python files

### PR DESCRIPTION
**Pull request by jdsika**
* Add .venv/ and venv/ to .gitignore
* Bump Python minimum to 3.8 as out of support
* Update GitHub Marketplace Actions
* Add workflow to check black formatting
* Add requirements_develop.txt
* Apply black formatting

### CODE REVIEW

File: format/OSITrace.py
Comment (PhRosenberger) [inline format/OSITrace.py:1]: If I see that correctly, this was only refactroing in this file, right?

File: format/osi2read.py
Comment (PhRosenberger) [inline format/osi2read.py:1]: This was also just refactored, right? - Why do you change all the '-signs to "-signs?

File: format/OSITrace.py
Comment (jdsika) [reply format/OSITrace.py:1]: Yes, I applied black formatting the same as it is applied in osi-validation. It is done in order to keep the same coding style throughout projects.

File: format/osi2read.py
Comment (jdsika) [reply format/osi2read.py:1]: This is the way

File: format/OSITrace.py
Comment (PhRosenberger) [reply format/OSITrace.py:1]: Ok fine, thx

**Review by PhRosenberger — APPROVED** [review]
I am fine with the code refactoring and black formatting.
I cannot help that much with the workflow changes.
Hope to get some more detailed feedback from @pmai there.

**Comment by pmai** [comment]
I have created a separate PR for non-python related build workflow changes, that also adds support for newer protobuf releases (upto current 25.x releases), leaving this PR for python-related changes to workflow and python code changes, if wanted.

The apt-get update dance should be dropped, since we should never depend on anything more specific from the base system.

I'm not a huge fan of the black reformatting (or usually any automatic reformattings), as it just seems to change one set of idiosynchracies for another, but whatever floats anyones boat (I think the code should be rewritten in places to make it clearer, something the reformatting really doesn't address).

**Comment by pmai** [comment]
I've refactored to built upon the other workflow enhancements, so once those are merged only the python-related changes will remain in this PR.

**Comment by jdsika** [comment]
> I have created a separate PR for non-python related build workflow changes, that also adds support for newer protobuf releases (upto current 25.x releases), leaving this PR for python-related changes to workflow and python code changes, if wanted.
> 
> The apt-get update dance should be dropped, since we should never depend on anything more specific from the base system.
> 
> I'm not a huge fan of the black reformatting (or usually any automatic reformattings), as it just seems to change one set of idiosynchracies for another, but whatever floats anyones boat (I think the code should be rewritten in places to make it clearer, something the reformatting really doesn't address).

not that I disagree with your stance towards formatting but in our case this will be done the same way as in the osi-validation and if a lot of people work on something it is helping to provide a consistent visual.

**Comment by pmai** [comment]
CCB 2014-01-15: Merge as-is.

**Pull request merged**